frontend/src/components/common/Pagination.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from "react";
+import {
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  Button,
+  NumberInput,
+  NumberInputField,
+  IconButton,
+  Flex,
+  Box,
+  Text,
+  GridItem,
+} from "@chakra-ui/react";
+
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronDownIcon,
+  TriangleDownIcon,
+} from "@chakra-ui/icons";
+import { getLocalStorageObjProperty } from "../../utils/LocalStorageUtils";
+
+type Props = {
+  numRecords: number;
+  resultsPerPage: number;
+  pageNum: number;
+  userPageNum: number;
+  setUserPageNum: React.Dispatch<React.SetStateAction<number>>;
+  setResultsPerPage: React.Dispatch<React.SetStateAction<number>>;
+  getLogRecords: (page_number: number) => Promise<void>;
+};
+
+const RESULTS_PER_PAGE_OPTIONS = [10, 25, 50];
+
+const Pagination = ({
+  numRecords,
+  resultsPerPage,
+  pageNum,
+  userPageNum,
+  setUserPageNum,
+  setResultsPerPage,
+  getLogRecords,
+}: Props): React.ReactElement => {
+  const numPages = Math.ceil(numRecords / resultsPerPage);
+
+  const handleNumberInputChange = (
+    newUserPageNumString: string,
+    newUserPageNum: number,
+  ) => {
+    setUserPageNum(newUserPageNum);
+    if (
+      newUserPageNum !== pageNum &&
+      newUserPageNum >= 1 &&
+      newUserPageNum <= numPages
+    ) {
+      getLogRecords(newUserPageNum);
```

Comment:
one issue I have with getting the new logs whenever the input changes is that the pagination gets really slow (I'm testing with 100000 logs) and its hard to tell when you're on the right page. I think it makes more sense to get the log records once the input is actually submitted. 

For example, if I start on page 1, and want to go to page 300, it'll fetch the logs for page 3, then page 30, then finally page 300. It should only retrieve the page 300 log records at the end, either by hitting enter on the input or clicking out of it. 

A lot of this headache is coming from the designs 😖 maybe we can approach the UI for this differently